src/server/game/Guilds/Guild.cpp

```diff
@@ -3330,12 +3344,15 @@ bool Guild::_ModifyBankMoney(CharacterDatabaseTransaction& trans, uint64 amount,
     return true;
 }
 
-void Guild::_SetLeaderGUID(Member* pLeader)
+void Guild::_SetLeader(CharacterDatabaseTransaction& trans, Member* pLeader)
 {
     if (!pLeader)
         return;
 
-    CharacterDatabaseTransaction trans = CharacterDatabase.BeginTransaction();
+    bool isInTransaction = bool(trans);
```

Comment:
I'd rather go with  trans != nullptr than casting like this.

Reply:
Doesn't `bool(trans)` convert to false if it is a nullptr by definition? 
https://stackoverflow.com/questions/17772103/can-i-use-if-pointer-instead-of-if-pointer-null/17772117#17772117